src/components/shared/Header.tsx

```diff
@@ -4,7 +4,8 @@ import Button from "./form/Button";
 import { statesModal } from "../../store/todoReducer";
 import { loginTodoAction } from "../../store";
 const Header = () => {
-  const { isLogedin, userName } = useSelector((state: statesModal) => state);
+  const { isLogedin } = useSelector((state: statesModal) => state);
+  const userName = window.localStorage.getItem("userName");
```

Comment:
We can keep the user info in state, we might need more info then userName only.
What do  you think?

Reply:
Updated